onecodex/viz/_custom_plots/__init__.py

```diff
@@ -86,6 +88,77 @@ def _convert_jsnull_to_none(obj: Any) -> Any:
     return obj
 
 
+async def _fetch_ndjson(url: str, headers: dict | None = None) -> list[dict[str, Any]]:
+    from pyodide.http import HttpStatusError
+
+    resp = await _fetch_with_retries(
+        url=url, headers=headers, retries=5, status_forcelist=(202, 429, 502, 503)
+    )
+
+    if resp.status == 202:
+        # Means data is not ready
+        raise HttpStatusError(status=202, status_text="Analysis results not ready", url=url)
+    resp.raise_for_status()
+
+    text = (
+        await resp.string()
+    )  # NDJSON (already decompressed if gzip since browsers do that automatically)
```

Comment:
What is NDJSON?

Reply:
Newline-delimitered JSON, it's this brand new and extraordinary idea of representing data - as JSONs delimitered with newlines 🤷